vnext/Microsoft.ReactNative.ComponentTests/CommonReaderTest.h

```diff
@@ -0,0 +1,324 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "winrt/Microsoft.ReactNative.h"
+
+namespace winrt::Microsoft::ReactNative::ReaderTestCases {
+
+// a construction to assert JSValueType::Object without hard-coded property order
+
+inline void TestCheckObject(std::map<hstring, std::function<void()>> props, IJSValueReader &reader) {
+  std::set<hstring> propNames;
+  hstring propertyName;
+
+  for (int i = 0; i < props.size(); i++) {
+    // ensure that we succeeded in reading a property name
+    TestCheckEqual(true, reader.GetNextObjectProperty(propertyName));
+
+    // ensure that we don't get duplicated property names from the reader
+    TestCheckEqual(propNames.end(), propNames.find(propertyName));
+    propNames.insert(propertyName);
+
+    // ensure that the property value is expected
+    auto it = props.find(propertyName);
+    TestCheckEqual(true, it != props.end());
```

Comment:
>TestCheckEqual [](start = 4, length = 14)

TestCheck? It ensures that the expression is true.

Reply:
Ah, I see that you use the TestCheckEqual everywhere for consistency. It makes sense.

---
In reply to: [432153602](https://github.com/microsoft/react-native-windows/pull/5046#discussion_r432153602) [](ancestors = 432153602)

Reply:
fixed